blockchain/chain_sync/src/tipset_syncer.rs

```diff
@@ -1283,21 +1283,22 @@ async fn validate_block<DB: Blockstore + Store + Clone + Sync + Send + 'static,
         .map_err(|e| (*block_cid, e.into()))?;
 
     // Async validations
-    let validations = FuturesUnordered::new();
+    let mut validations = JoinSet::new();
 
     // Check block messages
-    validations.push(tokio::task::spawn(check_block_messages::<_, C>(
+    let check_block_msgs = tokio::task::spawn(check_block_messages(
         Arc::clone(&state_manager),
         Arc::clone(&block),
         Arc::clone(&base_tipset),
-    )));
+    ));
+    validations.spawn(check_block_msgs);
 
     // Base fee check
     let smoke_height = state_manager.chain_config().epoch(Height::Smoke);
     let v_base_tipset = Arc::clone(&base_tipset);
     let v_block_store = state_manager.blockstore().clone();
     let v_block = Arc::clone(&block);
-    validations.push(tokio::task::spawn_blocking(move || {
+    let base_fee_check = tokio::task::spawn_blocking(move || {
```

Comment:
I'm not too fond of this pattern with the double spawns. A small helper function for waiting on blocking spawns might be prettier. Or we could create our own structure for having collections of validators. That way, `collect_errs` wouldn't be a free-standing function.